kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaCommandDecoder.java

```diff
@@ -294,11 +303,16 @@ protected void writeAndFlushResponseToClient(Channel channel) {
                         response.getRequest().getHeader());
                 }
 
-                ByteBuf result = responseToByteBuf(response.getResponseFuture().get(), response.getRequest());
+                ByteBuf result = responseToByteBuf(response.getResponseFuture().get(30,
+                        TimeUnit.SECONDS), response.getRequest());
                 channel.writeAndFlush(result);
             } catch (Exception e) {
-                // should not comes here.
-                log.error("error to get Response ByteBuf:", e);
+                // get response result may timeout.
+                log.error("request {}: error to get Response ByteBuf: ",
+                        response.getRequest().getHeader().apiKey(), e);
+                // make sure there is response to client.
+                responseQueue.add(response);
+                writeAndFlushWhenInactiveChannelOrTimeout(channel);
```

Comment:
Catch the specific exceptions to handle the timeout

```java
catch (ExecutionException | TimeoutException e) {
    // ...
} catch (Exception e) {
    // should not come here
```

Also, IMO, for the `TimeoutException`, we needn't to log the exception stack, but for the `ExecutionException`, we need to log `e.getCause()` instead of `e` itself.

See the documentation of `CompletableFuture#get`:

```java
    /**
     * Waits if necessary for at most the given time for this future
     * to complete, and then returns its result, if available.
     *
     * @param timeout the maximum time to wait
     * @param unit the time unit of the timeout argument
     * @return the result value
     * @throws CancellationException if this future was cancelled
     * @throws ExecutionException if this future completed exceptionally
     * @throws InterruptedException if the current thread was interrupted
     * while waiting
     * @throws TimeoutException if the wait timed out
     */
    public T get(long timeout, TimeUnit unit)
        throws InterruptedException, ExecutionException, TimeoutException {
```